### PR TITLE
refactor: golf `TileExistence.lean`

### PR DESCRIPTION
**Pull request by pitmonticone**

### CODE REVIEW

File: Carleson/TileExistence.lean [hunk 1]
```diff
@@ -649,8 +633,7 @@ lemma I3_prop_3_1 {k : ℤ} (hk : -S ≤ k) (y : Yk X k) :
   else
     rw [dif_neg hk_s]
     simp only [mem_preimage]
-    have : (y:X) ∈ ball o (4 * D^S-D^k:ℝ) := by
-      exact Yk_subset k y.property
+    have : (y:X) ∈ ball o (4 * D^S-D^k:ℝ) := Yk_subset k y.property
```
Comment (grunweg) [inline Carleson/TileExistence.lean:636]: If you want, I think mathlib style is to have spaces around binary operators.

**Review by grunweg — APPROVED** [review]
I know this repo doesn't require code review: I took a look anyway, essentially LGTM.

File: Carleson/TileExistence.lean [hunk 2]
```diff
@@ -156,14 +151,12 @@ lemma chain_property_set_has_bound (k : ℤ):
...
-      use z,hz
+      use z, hz
       specialize hc hz
       dsimp only [property_set] at hc
       rw [mem_setOf_eq] at hc
       exact hc.right.right r.left
-    intro h
-    left
-    exact h
+    · exact fun a_1 ↦ Or.intro_left (x ∈ if k = ↑S then {o} else ∅) a_1
```
[the remark this answers is not in the document]
Comment (pitmonticone) [reply Carleson/TileExistence.lean:159]: Done, thanks.

File: Carleson/TileExistence.lean [hunk 3]
```diff
@@ -595,12 +587,10 @@ mutual
...
       rw [Xk] at hx_mem_Xk
       simp only [mem_iUnion] at hx_mem_Xk ⊢
-      apply hx_mem_Xk.elim
-      intro y hy
+      refine hx_mem_Xk.elim (fun y hy ↦ ?_)
       use y
       rw [I3]
-      left
-      exact hy
+      exact mem_union_left (I2 hk y \ (Xk hk ∪ ⋃ y', ⋃ (_ : y' < y), I3 hk y')) hy
```
[the remark this answers is not in the document]
Comment (pitmonticone) [reply Carleson/TileExistence.lean:593]: Yes, it works. Done, thanks.

File: Carleson/TileExistence.lean (hunk 1 above)
Comment (pitmonticone) [reply Carleson/TileExistence.lean:636]: Yes, you're totally right. Done, thanks.

**Comment by pitmonticone** [comment]
I should have addressed all the review comments.

Thank you for your review, @grunweg.

**Comment by fpvandoorn** [comment]
Thanks! (And thanks for the review.)

**Pull request merged**